gtime/hierarchical/top_down.py

```diff
@@ -0,0 +1,178 @@
+
+       Parameters
+       ----------
+       model: BaseEstimator, required
+           time series forecasting model that is applied to each of the time series. A cross validation model
+           can also be passed.
+       hierarchy_tree: String or nx.DiGraph, optional, default: 'Infer'
+           Networkx Digraph containing the structure of the hierarchy tree.
+           If 'Infer' the selected root will have as many children as the remaining keys.
+        root: String, optional, default: first key of the dictionary of data
+        method: String, optional, default='tdsga'
+            Different top-down approaches are possible. Possible choices are:
+            -'tdsga': the proportion between the parent time series and the child is computed as the mean
+                      of every proportion at each timestep in the data provided.
+            -'tdsgf': the proportion between the parent time series and the child is computed as proportion
+                      of the means in the data provided.
+            -'tdfp':  Iterative method computing the proportions using the forecast of each
+                      time series as if it is indipendent.
+       Examples
+       --------
+       >>> import pandas._testing as testing
+       >>> from gtime.time_series_models import AR
+       >>> from gtime.hierarchical import HierarchicalTopDown
+       >>>
+       >>> testing.N, testing.K = 20, 1
+       >>>
+       >>> data1 = testing.makeTimeDataFrame(freq="s")
+       >>> data2 = testing.makeTimeDataFrame(freq="s")
+       >>> data3 = testing.makeTimeDataFrame(freq="s")
+       >>> data = {'data1': data1, 'data2': data2, 'data3' : data3}
+       >>>
+       >>> time_series_model = AR(p=2, horizon=3)
+       >>> tree_adj = {'data1': {'data2': {}, 'data3': {}}, 'data2': {}, 'data3': {}}
+       >>> tree = nx.DiGraph(tree_adj)
+       >>> hierarchical_model = HierarchicalTopDown(model=time_series_model, hierarchy_tree=tree, root='data1', method='tdsgf')
+       >>> hierarchical_model.fit(data)
+       >>> hierarchical_model.predict()
+       """
+    def __init__(self, model: BaseEstimator,
+                 hierarchy_tree: Union[str, nx.DiGraph] = "infer",
+                 root: str = None,
+                 method: str = 'tdsga'):
+        super().__init__(model=model, hierarchy_tree=hierarchy_tree)
+        self.root = root
+        self.method = method
+        self.proportions = {}
+
+    def fit(self, X: Dict[str, pd.DataFrame], y: pd.DataFrame = None):
+        """ Fit method
+
+        Parameters
+        ----------
+        X : Dict[str, pd.DataFrame], required
+            A dictionary of time series. Each is fitted independently
+        y : pd.DataFrame, optional, default = ``None``
+            only for compatibility
+
+        Returns
+        -------
+        self
+        """
+        self._check_is_dict_of_dataframes_with_str_key(X)
+        if self.root is None:
+            self.root = list(X.keys())[0]
```

Comment:
Ideally, this is used when the hierarchy tree is inferred and the choice is to have the first dataframe as root. Maybe we can do this only if we need to infer the hierarchy tree and otherwise look for the root in the tree?

Reply:
Yes, it would be better I think!